### ISSUE TITLE
RFE: please enable NVMe boot in OVMF (Bugzilla Bug 79)

### ISSUE THREAD

**Issue by tianocore-issues** [tianocore/edk2#6602]
This issue was created automatically with [bugzilla2github](https://github.com/makubacki/bugzilla2github/tree/bz2gh_tianocore)
# Bugzilla Bug 79
Date: 2016-07-21T15:04:21+00:00
From: @lersek
To:   @lersek
CC:   @lersek

See also:     https://github.com/tianocore/edk2/issues/48
Last updated: 2016-07-21T15:23:07+00:00

**Comment by tianocore-issues** [comment]
## Comment 239
Date: 2016-07-21 15:04:37 +0000
From: @lersek

lersek commented on Jan 26

What does NVMe enable that virtio-blk / virtio-scsi disks don&apos;t?

(Not arguing against the feature request, I&apos;d just like to know.)

Please also provide a reasonably short QEMU command line that exposes a drive as an NVMe device to the guest. Thanks.

**Comment by tianocore-issues** [comment]
## Comment 241
Date: 2016-07-21 15:05:03 +0000
From: @lersek

stamerlan changed the title from NVMe boot to OVMF on Jan 26

**Comment by tianocore-issues** [comment]
## Comment 242
Date: 2016-07-21 15:06:55 +0000
From: @lersek

lersek commented on Jan 26

@stamerlan haha, sorry, I was not clear enough. In more details:

This tracker is not just for OVMF, it is for the entirety of edk2. In sensible bug tracking systems, like Bugzilla, you have products, components, sub-components and such -- metadata for each bug report. When you report a bug in those sensible systems, you are expected to make an effort to categorize your bug report up-front, in order to save time for the people who read the bug reports.

In this bug tracker though, we have no such metadata, the best we can do is (apparently) &quot;tags&quot;. So @jljusten tends to tag OVMF-related bug reports with the &quot;OVMF&quot; label. But, since you know up-front that you are reporting the issue for OVMF, not another platform built from edk2, you should immediately signal in the title of your report that it is for OVMF.

Therefore the title should contain both OVMF and NVMe boot.

Really, there&apos;s a huge disconnect here between users of OVMF, and edk2, the open source project. One advantage that a well-maintained Bugzilla instance has is that it forces bug reporters to at least think about metadata, plus it provides a bug report template that must be completed with important information. Github&apos;s issue tracker encourages reckless user behavior -- just dump the three words that come to your mind into a new item. (But hey at least you can format those three words with markdown!)

This open letter is completely justified: https://github.com/dear-github/dear-github

To top it all off, I personally lack the credentials to change bug titles, bug statuses, tags; anything at all beyond commenting. (Except, of course, I have the privilege of fixing bugs.)

I knew using a web-era issue tracker for edk2 would be a mistake, and I stand vindicated.

TL;DR: please change the title to: &quot;RFE: please enable NVMe boot in OVMF&quot;.

Also, please try to answer my other questions above [1]. In any RFE (request for enhancement), you gotta come up with some justification why the developers should spend their time on the feature you&apos;d like to have. As I said, I don&apos;t doubt this is a reasonable request, but I do care about your reasons for wanting to use NVMe.

[1] In a sane bug tracker, I would reference earlier comments by writing &quot;comment 3&quot; or &quot;comment 4&quot;, and the tracker would automatically turn those words into links. In this tracker however, comments have no numbers (they don&apos;t even have human readable exact timestamps! WTF is &quot;lersek commented an hour ago&quot; supposed to mean?!), I can only copy and paste garbage-looking links like this: #48 (comment) &lt;https://github.com/tianocore/edk2/issues/48#issuecomment-175144378&gt; [--&gt; comment 1 in Bugzilla]

GitHub issue tracker, the social media of bug reporting, for the facebook generation.

**Comment by tianocore-issues** [comment]
## Comment 243
Date: 2016-07-21 15:08:08 +0000
From: @lersek

stamerlan commented on Jan 26

I meant qemu virtual nvme disk:
qemu-system-x86_64 \
-drive file=ssd.img,if=none,id=disk0,format=raw \
-device nvme,drive=drive0,serial=1234

**Comment by tianocore-issues** [comment]
## Comment 244
Date: 2016-07-21 15:08:22 +0000
From: @lersek

stamerlan changed the title from &quot;OVMF&quot; to &quot;RFE: please enable NVMe boot in OVMF&quot; on Jan 26

**Comment by tianocore-issues** [comment]
## Comment 245
Date: 2016-07-21 15:08:34 +0000
From: @lersek

lersek commented on Jan 26

I will assume that http://thread.gmane.org/gmane.comp.bios.coreboot.seabios/10375 is related.

**Comment by tianocore-issues** [comment]
## Comment 246
Date: 2016-07-21 15:08:46 +0000
From: @lersek

stamerlan commented on Jan 26

Yup, @lersek it&apos;s my e-mail. Shell I add explanation in the 1st post or leave one more comment?

**Comment by tianocore-issues** [comment]
## Comment 247
Date: 2016-07-21 15:08:57 +0000
From: @lersek

lersek commented on Jan 26

@stamerlan if you could expand the first comment with details, that would be great. Thanks!

**Comment by tianocore-issues** [comment]
## Comment 248
Date: 2016-07-21 15:09:12 +0000
From: @lersek

lersek commented on Jan 26

@stamerlan thanks for explaining your use case in the first comment. It is perfectly reasonable; OVMF is regularly used for prototyping and testing otherwise hardware-oriented features.

**Comment by tianocore-issues** [comment]
## Comment 249
Date: 2016-07-21 15:11:24 +0000
From: @lersek

stamerlan commented on Jan 26

@lersek as far as I understand virtio parameters used to cooperate with hypervisor to use real HW?

**Comment by tianocore-issues** [comment]
## Comment 250
Date: 2016-07-21 15:11:36 +0000
From: @lersek

lersek commented on Jan 26

@stamerlan I don&apos;t understand the question. But anyway I just wanted to be sure that NVMe was specifically needed for your use case (i.e., it couldn&apos;t be served by virtio disks).

**Comment by tianocore-issues** [comment]
## Comment 251
Date: 2016-07-21 15:11:49 +0000
From: @lersek

lersek commented on Jan 27

Posted prerequisite patch for QEMU:
http://news.gmane.org/find-root.php?message_id=1453850483-27511-1-git-send-email-lersek@redhat.com

**Comment by tianocore-issues** [comment]
## Comment 252
Date: 2016-07-21 15:13:20 +0000
From: @lersek

lersek added a commit to lersek/edk2 that referenced this issue on Jan 27
https://github.com/tianocore/edk2/issues/48#ref-commit-d354ec6
OvmfPkg: include NvmExpressDxe driver
https://github.com/lersek/edk2/commit/d354ec63665764aabf4cdd2f029faa4be117df39

lersek added a commit to lersek/edk2 that referenced this issue on Jan 27
https://github.com/tianocore/edk2/issues/48#ref-commit-2d3a507
OvmfPkg: QemuBootOrderLib: recognize NVMe devices
https://github.com/lersek/edk2/commit/2d3a507ef13500d0d727022fedeee4418b497740

**Comment by tianocore-issues** [comment]
## Comment 253
Date: 2016-07-21 15:13:32 +0000
From: @lersek

lersek commented on Jan 27

Posted edk2 series:
http://thread.gmane.org/gmane.comp.bios.edk2.devel/6950

**Comment by tianocore-issues** [comment]
## Comment 254
Date: 2016-07-21 15:14:56 +0000
From: @lersek

lersek commented on Jan 27

@stamerlan: can you please test my patches (QEMU and OVMF) that I linked in [bugzilla comment 13] and [bugzilla comment 15]? I CC&apos;d you on both sets directly, so you should have them in your email. A Tested-by from users is highly appreciated. Thanks.

**Comment by tianocore-issues** [comment]
## Comment 255
Date: 2016-07-21 15:15:25 +0000
From: @lersek

stamerlan commented on Jan 28

@Hi, @lersek
I saw e-mails, but I got sick, so I&apos;ll come to office to test it tomorrow.

**Comment by tianocore-issues** [comment]
## Comment 256
Date: 2016-07-21 15:15:35 +0000
From: @lersek

lersek commented on Jan 28

@stamerlan take your time, get better first. thx

**Comment by tianocore-issues** [comment]
## Comment 257
Date: 2016-07-21 15:15:51 +0000
From: @lersek

stamerlan commented on Jan 29

Hi, @lersek

Yes, your patch is working. I successfully installed windows 10 x86_64 on qemu!

Thank you a lot!

P.S. what happened with nvme_boot_issue48, today I can&apos;t checkout it?

**Comment by tianocore-issues** [comment]
## Comment 258
Date: 2016-07-21 15:16:02 +0000
From: @lersek

@stamerlan stamerlan closed this on Jan 29

**Comment by tianocore-issues** [comment]
## Comment 259
Date: 2016-07-21 15:16:36 +0000
From: @lersek

lersek commented on Jan 29

@stamerlan Thank you very much for the testing and the feedback. I highly appreciate it!

Please don&apos;t close this item just yet though -- can you please reopen it? The patches are still under review, and have not been committed to either QEMU or OVMF. This item should only be closed when the edk2 patches are &quot;in&quot;.

However, please do report your successful testing in both the QEMU and edk2 threads. Please locate the following messages in your mailbox:

[PATCH 0/3] OvmfPkg: enable NVMe devices
[PATCH] nvme: generate OpenFirmware device path in the &quot;bootorder&quot; fw_cfg file

hit Reply All, and respond with

Tested-by: Vladislav Vovchenko &lt;vladislav.vovchenko@sk.com&gt;

This way reviewers will see that the patch has undergone independent testing, and your testing feedback will be captured in the commit log. Thank you!

**Comment by tianocore-issues** [comment]
## Comment 260
Date: 2016-07-21 15:17:06 +0000
From: @lersek

lersek commented on Jan 29

@stamerlan

&gt; P.S. what happened with nvme_boot_issue48, today I can&apos;t checkout it?

I have no clue, it must be a github issue or a network glitch between you &amp; github; I didn&apos;t touch this branch.

(I never touch branches that I push for review. If a new version of the series becomes necessary, I push XXXX_v2, XXXX_v3 branches. Such a branch should stay reflecting the mailing list posting.)

**Comment by tianocore-issues** [comment]
## Comment 261
Date: 2016-07-21 15:17:14 +0000
From: @lersek

jljusten reopened this on Jan 29

**Comment by tianocore-issues** [comment]
## Comment 262
Date: 2016-07-21 15:17:22 +0000
From: @lersek

jljusten added the OVMF label on Jan 29

**Comment by tianocore-issues** [comment]
## Comment 263
Date: 2016-07-21 15:17:40 +0000
From: @lersek

stamerlan commented on Feb 1


Sorry for my delay. I have no access to my work-mailbox from home.
I replied today in qemu and edk2 threads.

Thank you a lot!

**Comment by tianocore-issues** [comment]
## Comment 264
Date: 2016-07-21 15:19:03 +0000
From: @lersek

qemu-deploy pushed a commit to qemu/qemu that referenced this issue on Feb 2
https://github.com/tianocore/edk2/issues/48#ref-commit-a907ec5
nvme: generate OpenFirmware device path in the &quot;bootorder&quot; fw_cfg file
https://github.com/qemu/qemu/commit/a907ec52cc1aefc820768b6e341b56f8f3caaca7

jljusten pushed a commit that referenced this issue on Feb 2
https://github.com/tianocore/edk2/issues/48#ref-commit-8ae3832
OvmfPkg: include NvmExpressDxe driver
https://github.com/tianocore/edk2/commit/8ae3832df9985f5285aa59102c94d14ba952844b

jljusten pushed a commit that referenced this issue on Feb 2
https://github.com/tianocore/edk2/issues/48#ref-commit-d7c0dfa
OvmfPkg: QemuBootOrderLib: recognize NVMe devices
https://github.com/tianocore/edk2/commit/d7c0dfaef26ca1693529360ed6c6de0021e19595

**Comment by tianocore-issues** [comment]
## Comment 265
Date: 2016-07-21 15:19:36 +0000
From: @lersek

lersek commented on Feb 3

Patches are upstream, both QEMU and edk2. Closing.

lersek closed this on Feb 3

**Comment by tianocore-issues** [comment]
## Comment 266
Date: 2016-07-21 15:20:44 +0000
From: @lersek

mdkinney added a commit to mdkinney/edk2 that referenced this issue on Mar 9
https://github.com/tianocore/edk2/issues/48#ref-commit-8f6dcc8
OvmfPkg: include NvmExpressDxe driver
https://github.com/mdkinney/edk2/commit/8f6dcc82d2532eb30fe927ea97094504de030e9b

mdkinney added a commit to mdkinney/edk2 that referenced this issue on Mar 9
https://github.com/tianocore/edk2/issues/48#ref-commit-2b52e05
OvmfPkg: QemuBootOrderLib: recognize NVMe devices 
https://github.com/mdkinney/edk2/commit/2b52e054f89bb73ba0d3f7517c51deb00d3f99c9

**Comment by tianocore-issues** [comment]
## Comment 267
Date: 2016-07-21 15:23:07 +0000
From: @lersek

lersek referenced this issue on May 9

RFE: support USB Attached Storage #85
https://github.com/tianocore/edk2/issues/85
[--&gt; bugzilla: see bug 53]
